docs/config.md

```diff
@@ -0,0 +1,156 @@
+# Talos CCM Configuration
+
+## Overview
+
+The Talos CCM is a Kubernetes controller that manages the lifecycle of Talos nodes in a Kubernetes cluster.
+
+In scope of the Talos CCM, the following features are implemented:
+* Initialize nodes in a Kubernetes cluster, set [well-known labels and annotations](https://kubernetes.io/docs/reference/labels-annotations-taints/).
+* Label and annotate nodes based on the transformation rules.
+* Approve and sign the [kubelet certificate signing request](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#kubernetes-signers) for Talos nodes.
+
+Result of kubernetes node object after Talos CCM initialization:
+
+```yaml
+apiVersion: v1
+kind: Node
+metadata:
+  annotations:
+    ...
+    # Set annotations based on transformation rules
+    talos.dev/instance-id: "id-e8e8c388-5812-4db0-87e2-ad1fee51a1c1"
+    ...
+  labels:
+    ...
+    # Set well-known labels
+    node.cloudprovider.kubernetes.io/platform: metal
+    node.kubernetes.io/instance-type: t2.micro
+    topology.kubernetes.io/region: us-west-1-on-metal
+    topology.kubernetes.io/zone: us-west-1f
+    # Label based on transformation rules
+    node-role.kubernetes.io/web: ""
+    ...
+  name: web-1
+spec:
+  ...
+  # Define the provider ID of the node
+  providerID: someproviderID:///e8e8c388-5812-4db0-87e2-ad1fee51a1c1
+status:
+  # Define the addresses of the node
+  addresses:
+  - address: 172.16.0.11
+    type: InternalIP
+  - address: 1.2.3.4
+    type: ExternalIP
+  - address: 2001:123:123:123::1
+    type: ExternalIP
+  - address: web-1
+    type: Hostname
+```
+
+## Configuration
+
+```yaml
+# Global parameters
+global:
+  # Check and approve node client certificate signing requests
+  # In case when you use `rotate-server-certificates` flag in kubelet
+  # Parameters are optional, by default is "false"
+  approveNodeCSR: true
```

Comment:
Will this controller break if the value is left false?

Reply:
It is optional param. If false - you need to approve CSR manually or another controller.